nipoppy/cli.py

```diff
@@ -0,0 +1,259 @@
+    help=(
+        "Set all statuses to False in newly added records"
+        " (regardless of what is on disk). May be useful to reduce runtime."
+    ),
+)
+@click.option(
+    "--regenerate",
+    "--force",
+    "-f",
+    is_flag=True,
+    help=(
+        "Regenerate the doughnut file even if it already exists"
+        " (default: only append rows for new records)"
+    ),
+)
+@global_options
+def doughnut(**params):
+    """Create or update a dataset's doughnut file."""
+    from nipoppy.workflows.doughnut import DoughnutWorkflow
+
+    with handle_exception(DoughnutWorkflow(**params)) as workflow:
+        workflow.run()
+
+
+@cli.command()
+@dataset_option
+@click.option(
+    "--copy-files",
+    is_flag=True,
+    help="Copy files when reorganizing (default: create symlinks).",
+)
+@click.option(
+    "--check-dicoms",
+    is_flag=True,
+    help=(
+        "Read DICOM file headers when reorganizing and check if they have the "
+        '"DERIVED" image type (which can be problematic for some BIDS '
+        "converters). The paths to the derived DICOMs will be written to the log."
+    ),
+)
+@global_options
+def reorg(**params):
+    """(Re)organize raw (DICOM) files.
+
+    From ``<DATASET_ROOT>/sourcedata/imaging/pre_reorg`` to
+    ``<DATASET_ROOT>/sourcedata/imaging/post_reorg``
+    """
+    from nipoppy.workflows.dicom_reorg import DicomReorgWorkflow
+
+    with handle_exception(DicomReorgWorkflow(**params)) as workflow:
+        workflow.run()
+
+
+@cli.command()
+@dataset_option
+@runners_options
+@global_options
+def bidsify(**params):
+    """Run a BIDS conversion pipeline."""
+    from nipoppy.workflows.bids_conversion import BidsConversionRunner
+
+    with handle_exception(BidsConversionRunner(**params)) as workflow:
+        workflow.run()
+
+
+@cli.command()
+@dataset_option
+@runners_options
+@click.option(
+    "--keep-workdir",
+    is_flag=True,
+    help=(
+        "Keep pipeline working directory upon success "
+        "(default: working directory deleted unless a run failed)"
+    ),
+)
+@click.option(
+    "--tar",
+    is_flag=True,
+    help=(
+        "Archive participant-session-level results into a tarball upon "
+        "successful completion. The path to be archived should be specified "
+        "in the tracker configuration file."
+    ),
+)
+@global_options
+def run(**params):
+    """Run a processing pipeline."""
+    from nipoppy.workflows.runner import PipelineRunner
+
+    with handle_exception(PipelineRunner(**params)) as workflow:
+        workflow.run()
+
+
+@cli.command()
+@dataset_option
+@pipeline_options
+@global_options
+def track(**params):
+    """Track the processing status of a pipeline."""
+    from nipoppy.workflows.tracker import PipelineTracker
+
+    with handle_exception(PipelineTracker(**params)) as workflow:
+        workflow.run()
+
+
+@cli.command()
+@dataset_option
+@runners_options
+@global_options
+def extract(**params):
+    """Add subparser for extract command."""
```

Comment:
```suggestion
    """Run an extraction pipeline."""
```
or something equally/more informative